Lib/asyncio/base_events.py

```diff
@@ -839,12 +874,49 @@ def _check_sendfile_params(self, sock, file, offset, count):
                 "offset must be a non-negative integer (got {!r})".format(
                     offset))
 
+    async def _connect_sock(self, exceptions, addr_info, local_addr_infos=None):
+        """Create, bind and connect one socket."""
+        my_exceptions = []
+        exceptions.append(my_exceptions)
+        family, type_, proto, _, address = addr_info
+        sock = None
+        try:
+            sock = socket.socket(family=family, type=type_, proto=proto)
+            sock.setblocking(False)
+            if local_addr_infos is not None:
+                for _, _, _, _, laddr in local_addr_infos:
+                    try:
+                        sock.bind(laddr)
+                        break
+                    except OSError as exc:
+                        msg = (
+                            f'error while attempting to bind on '
+                            f'address {laddr!r}: '
+                            f'{exc.strerror.lower()}'
+                        )
+                        exc = OSError(exc.errno, msg)
+                        my_exceptions.append(exc)
+                else:  # all bind attempts failed
+                    raise my_exceptions.pop()
+            await self.sock_connect(sock, address)
+            return sock
+        except OSError as exc:
+            my_exceptions.append(exc)
+            if sock is not None:
+                sock.close()
+            raise
+        except:
+            if sock is not None:
+                sock.close()
+            raise
+
     async def create_connection(
             self, protocol_factory, host=None, port=None,
             *, ssl=None, family=0,
             proto=0, flags=0, sock=None,
             local_addr=None, server_hostname=None,
-            ssl_handshake_timeout=None):
+            ssl_handshake_timeout=None,
+            delay=None, interleave=None):
```

Comment:
Need to mention these in the docstring.  

Is it a good idea to have one parameter instead of two?  For example, `interleave=-1` (default) is used to disable happy eyeballs. `interleave>0` is used as a delay for happy eyeballs.

Reply:
> Why shouldn't we always interleave addrs?

Mostly because RFC8305 specifies "First Address Family Count" as a configurable value, and since I'm making it configurable, might as well provide the option of turning it off. Also, address family interleaving is a "SHOULD" in the RFC. If `interleave` is not specified, the default behavior is to interleave.

If we decide that the use case for configuring `interleave` is too small, we could remove it outright and just always interleave.

Reply:
> Need to mention these in the docstring.

The docstring of `create_connection` itself? The existing docstring doesn't explain all those old arguments, so I wondered whether it's appropriate to mention the new one.

(The larger Python docs should probably be updated as well?)